pkg/management/authorizationdiff.go

```diff
@@ -0,0 +1,98 @@
+package management
+
+import (
+	"github.com/cloudtrust/common-service/v2/configuration"
+)
+
+const (
+	nilString = "<<null>>"
+)
+
+// AlterFunction type
+type AlterFunction func(configuration.Authorization)
+
+// AuthorizationSet interface
+type AuthorizationSet interface {
+	apply(other AuthorizationSet, addFunction AlterFunction, removeFunction AlterFunction)
+}
+
+type diffStatus struct {
+	value configuration.Authorization
+}
+
+type authorizationSet struct {
+	authz map[string]any
```

Comment:
as we are dealing with authorization, do we really need a generic map ?

Reply:
It allows to write recursive generic code rather than code specific to each level of mapping. It was first written in a non-generic way and required almost 5 times more lines of code, just for copied/pasted code with a very small change at each time. For unit tests, it would have been necessary to write 5 times more tests.